docs/spec/crosschain/messages.md

```diff
@@ -132,13 +151,43 @@ is the first vote, a new ballot is created. When a threshold of votes is
 reached, the ballot is finalized. When a ballot is finalized, a new CCTX is
 created.
 
-If the receiver chain is a ZetaChain, the EVM deposit is handled and the
-status of CCTX is changed to "outbound mined". If EVM deposit handling fails,
-the status of CCTX is chagned to 'aborted'.
+If the receiver chain is ZetaChain, `HandleEVMDeposit` is called. If the
+tokens being deposited are ZETA, `MintZetaToEVMAccount` is called and the
+tokens are minted to the receiver account on ZetaChain. If the tokens being
+deposited are gas tokens or ERC20 of a connected chain, ZRC20's `deposit`
+method is called and the tokens are deposited to the receiver account on
+ZetaChain. If the message is not empty, system contract's `depositAndCall`
+method is also called and an omnichain contract on ZetaChain is executed.
+Omnichain contract address and arguments are passed as part of the message.
+If everything is successful, the CCTX status is changed to `OutboundMined`.
+
+If the receiver chain is a connected chain, the `FinalizeInbound` method is
+called to prepare the CCTX to be processed as an outbound transaction. To
+cover the outbound transaction fee, the required amount of tokens submitted
+with the CCTX are swapped using a Uniswap pool on ZetaChain for the ZRC20 of
```

Comment:
Should we call it a Uniswap pool? It is a Uniswap V2 contract instance deployed on ZetaChain

Reply:
5ae235a5